pinot-core/src/main/java/org/apache/pinot/core/query/selection/SelectionOperatorService.java

```diff
@@ -104,13 +105,20 @@ private Comparator<Object[]> getTypeCompatibleComparator(List<OrderByExpressionC
 
     int numValuesToCompare = valueIndexList.size();
     int[] valueIndices = new int[numValuesToCompare];
-    boolean[] isNumber = new boolean[numValuesToCompare];
+    boolean[] useDoubleComparison = new boolean[numValuesToCompare];
+    boolean[] useBigDecimalComparison = new boolean[numValuesToCompare];
```

Comment:
We don't need to track this flag. `BigDecimal` can reuse the general `Comparable` comparison